contracts/src/TeleporterTokenSource.sol

```diff
@@ -412,7 +412,7 @@ abstract contract TeleporterTokenSource is
                 abi.decode(bridgeMessage.payload, (SingleHopCallMessage));
 
             uint256 sourceAmount =
-                _processReceivedTransfer(sourceBlockchainID, originSenderAddress, payload.amount);
+                _processSingleHopTransfer(sourceBlockchainID, originSenderAddress, payload.amount);
 
             // Verify that the payload's source blockchain ID matches the source blockchain ID passed from Teleporter.
             // Prevents a destination bridge from accessing tokens attributed to another destination bridge instance.
```

Comment:
I know it wasn't introduced in this PR, but I think we should update this comment in a later PR. Just noting it for future visability.

The check prevents a destination from forging an incorrect `sourceBlockchainID` in the call to `receiveTokens` in the send and call case, but it is not related to accessing the tokens attributed to other destination instances (since that is all authenticated by the Teleporter sender directly).